app/javascript/utilities/hooks/useGlobalKeyEventListener.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'preact/hooks';
+
+const FOCUSED_TAG_EXCLUDE_LIST = ['INPUT', 'TEXTAREA'];
+
+export const registerGlobalKeyEventListener = (keys, callback) => {
+  if (!keys || keys.length === 0 || !callback) {
+    return null;
+  }
+
+  const eventListener = (event) => {
+    const { tagName, classList } = document.activeElement;
+
+    if (
+      !keys.includes(event.key) ||
+      FOCUSED_TAG_EXCLUDE_LIST.includes(tagName) ||
+      classList.contains('input')
+    )
+      return;
+
+    callback(event);
+  };
+
+  window.addEventListener('keydown', eventListener);
+
+  return eventListener;
+};
+
+export default (keys, callback) => {
```

Comment:
syntax (non-blocking): It's not documented in the project, but `default` exports are usually not preferred as it can prevent refactoring via an editor like VS Code. For more context, see the following posts.

* [Why I've stopped exporting defaults from my JavaScript modules](https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module)
* [Avoid Export Default](https://basarat.gitbook.io/typescript/main-1/defaultisbad)